Branch-SDK/src/io/branch/referral/BranchPreinstall.java

```diff
@@ -0,0 +1,105 @@
+package io.branch.referral;
+
+import android.content.Context;
+import io.branch.referral.Defines.PreinstallKey;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by --vbajpai on --24/07/2019 at --13:44 for --android-branch-deep-linking-attribution
+ */
+class BranchPreinstall {
+
+  private String SYSTEM_PROPERTIES_CLASS_KEY = "android.os.SystemProperties";
+  private String BRANCH_PREINSTALL_PROP_KEY = "ro.branch.preinstall.apps.path";
+
+  public void getPreinstallSystemData(Context context) {
```

Comment:
Again, this should be static, and probably take a `Branch` as an argument

Reply:
Done